src/test/run-pass/rfc-1937-termination-trait/termination-trait-in-test.rs

```diff
@@ -26,6 +26,7 @@ fn is_a_num() -> Result<(), ParseIntError> {
 #[test]
 #[should_panic]
 fn not_a_num() -> Result<(), ParseIntError> {
+    //~^ ERROR test functions returning Result<_, _> must not use #[should_panic]
```

Comment:
This test livs in the `run-pass` directory -- that means it must compile. Can you copy the test into `ui/rfc-1937-termination-trait/`, and break it into two files? One of them can include the `is_a_num` test (and the comment `// run-pass`). The other includes the `not_a_num` test (with the `//~ ERROR`). That way, we test that the first test compiles and runs, and that the second fails to compile. There is discussion about ui tests in the rustc-guide: https://rust-lang-nursery.github.io/rustc-guide/tests/adding.html#ui